aspects/aspects.go

```diff
@@ -331,6 +331,42 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 	return notFoundErrorFrom(a, request, "no matching write rule")
 }
 
+// namespaceResult creates a nested namespace around the result that corresponds
+// to the unmatched entry parts. Unmatched placeholders are filled in using maps
+// of all the matching values in the databag.
+func namespaceResult(res interface{}, suffixParts []string) (interface{}, error) {
+	if len(suffixParts) == 0 {
+		return res, nil
+	}
+
+	part := suffixParts[0]
+	if part[0] == '{' && part[len(part)-1] == '}' {
```

Comment:
I think it should be covered by [validateAspectDottedPath](https://github.com/snapcore/snapd/blob/master/aspects/aspects.go?plain=1#L254), but might still be worth it to check the length just in case.

Reply:
You found the answer already but yes, these rules come from an assertion which we validate when processing (and it will likely have some basic checks when building it as well, although that part hasn't been done yet)